### PR TITLE
br: fix S3 backup endpoint suffix

### LINKED ISSUE #30104: br s3 endpoint can't end with '/'
## Bug Report

Please answer these questions before submitting your issue. Thanks!

This bug is a little bit frustrating. 

### 1. Minimal reproduce step (Required)

```
AWS_ACCESS_KEY_ID=xxxx AWS_SECRET_ACCESS_KEY=xxxxx tiup br backup full --s3.endpoint http://minio.pingcap.net:9000/ -s s3://tmp/sysbenc-empty-cy
```

### 2. What did you expect to see? (Required)
br backup successfully

### 3. What did you see instead (Required)
br backup failed
```
Error: error happen in store 1 at 127.0.0.1:20160: Io(Custom { kind: Other, error: "failed to put object Request ID: None Body: <?xml version=\"1.0\" encoding=\"UTF-8\"?>\n<Error><Code>BadRequest</Code><Message>An error occurred when parsing the HTTP request PUT at &#39;/
/tmp/sysbenc-empty-cy/1_2_28_2550d727ffe3e408af41f541b134d5e064440c25988bbecfb665933d1195a45d_1637743793014_write.sst&#39;</Message><Resource>//tmp/sysbenc-empty-cy/1_2_28_2550d727ffe3e408af41f541b134d5e064440c25988bbecfb665933d1195a45d_1637743793014_write.sst</Resource><
RequestId></RequestId><HostId>bf48f99f-3f89-4a20-bd11-95d1658fa9c8</HostId></Error>" }): [BR:KV:ErrKVStorage]tikv storage occur I/O error
Error: run `/root/.tiup/components/br/v5.3.0/br` (wd:/root/.tiup/data/Spfa3FJ) failed: exit status 1
```

### 4. What is your TiDB version? (Required)

**5.3.0**

### LINKED ISSUE #30128: During S3 backup, errors may occur if there is "/" at the end of the endpoint
Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.

when S3 backup or restore, If "/" is added last in the endpoint, an error may occur

2. What did you expect to see?

Whether "/" is added last, the backup can be successful

3. What did you see instead?

4. What version of BR and TiDB/TiKV/PD are you using?

br -4.0.11
br -5
The above problem occurs only when there is data in instance. Therefore, when br version is 4.0.11, this problem will not occur when backing up empty instances, because the system library will not be backed up; However, when the br version is 5, even for empty instances, Br the system library will be backed up during backup.

5. Operation logs
   - Please upload `br.log` for BR if possible
   - Please upload `tidb-lightning.log` for TiDB-Lightning if possible
   - Please upload `tikv-importer.log` from TiKV-Importer if possible
   - Other interesting logs

6. Configuration of the cluster and the task
   - `tidb-lightning.toml` for TiDB-Lightning if possible
   - `tikv-importer.toml` for TiKV-Importer if possible
   - `topology.yml` if deployed by TiUP

7. Screenshot/exported-PDF of Grafana dashboard or metrics' graph in Prometheus if possible

### PR DESCRIPTION
**Pull request by tongtongyin**
### What problem does this PR solve?

Issue Number: close #30104

Problem Summary:

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by kennytm** [comment]
as mentioned before in https://github.com/pingcap/tidb/pull/30144#pullrequestreview-818687216, fixing the slash in `options.Endpoint` won't be effective when the endpoint is defined as part of the `s3://` url.

**Comment by tongtongyin** [comment]
As mentioned before in #30144 (review), It will also report an error.



As long as there is "/" at the end of the endpoint parameter, both methods will report errors.

**Comment by tongtongyin** [comment]
I submit and update the code, and now I can guarantee that these two backup methods can succeed.
eg. --endpoint http://ip:port/
eh. --storage "endpoint＝http://ip:port/"

These two ways can be success backup whether in the back of endpoint add "/".

File: br/pkg/storage/parse.go [hunk 1]
```diff
@@ -140,7 +140,7 @@ func ExtractQueryParameters(u *url.URL, options interface{}) {
 					field.SetBool(v)
 				}
 			case reflect.String:
-				field.SetString(param)
+				field.SetString(strings.TrimSuffix(param, "/"))
```
Comment (kennytm) [inline br/pkg/storage/parse.go:143]: 😅

this code is also used by other string fields like `secret-access-key` so we can't place the `TrimSuffix` here.

Comment (tongtongyin) [reply br/pkg/storage/parse.go]: It's embarrassing. Please /cc @kennytm again

**Review by kennytm — APPROVED** [review]

**Review by glorv — APPROVED** [review]

**Comment by kennytm** [comment]
```
[2021-12-20T06:10:21.971Z] FAIL: pessimistic_test.go:488: testPessimisticSuite.TestSelectForUpdateNoWait

[2021-12-20T06:10:21.971Z] 

[2021-12-20T06:10:21.972Z] pessimistic_test.go:566:

[2021-12-20T06:10:21.972Z]     c.Check(err, NotNil)

[2021-12-20T06:10:21.972Z] ... value = nil
```

/run-check_dev_2

**Pull request merged**